### PR TITLE
feat: perform a segue from Viewcontroller to DetailMovieVC, populate …

### PR DESCRIPTION
**Pull request by homerblr**
Task #14 WIP. Need to populate countryAndLength label with info (currently no such data at [MovieDetails] class), probably should put numberFormat and dateFormat methods in another Class

### CODE REVIEW

File: Cinema Journal/Controller/ViewController.swift [hunk 1]
```diff
@@ -51,24 +61,41 @@ extension ViewController : UITableViewDataSource {
...
+        performSegue(withIdentifier: segueID, sender: indexPath)
+        tableView.deselectRow(at: indexPath, animated: true)
     }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+         if segue.identifier == segueID {
+             guard let destinationVC = segue.destination as? DetailMovieVC else {return}
+            let row = (sender as! NSIndexPath).row
+            let model = results[row]
+            DispatchQueue.main.async {
```
Comment (iKisliy) [inline Cinema Journal/Controller/ViewController.swift]: Segue perfomed always on main queue, dispatch is redundant

File: Cinema Journal/Controller/ViewController.swift [hunk 2]
```diff
@@ -51,24 +61,41 @@ extension ViewController : UITableViewDataSource {
...
+extension ViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        performSegue(withIdentifier: segueID, sender: indexPath)
+        tableView.deselectRow(at: indexPath, animated: true)
     }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+         if segue.identifier == segueID {
+             guard let destinationVC = segue.destination as? DetailMovieVC else {return}
+            let row = (sender as! NSIndexPath).row
```
Comment (iKisliy) [inline Cinema Journal/Controller/ViewController.swift]: place inside guard

File: Cinema Journal/View/detailMovieVC.swift [hunk 3]
```diff
@@ -22,20 +23,58 @@ class detailMovieVC: UIViewController {
...
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
     }
-    */
+    
+    func configureModel (_ model: MovieDetails) {
```
Comment (iKisliy) [inline Cinema Journal/View/detailMovieVC.swift]: hate that space between function and parenthesis

File: Cinema Journal/Controller/ViewController.swift (hunk 1 above)
Comment (homerblr) [reply Cinema Journal/Controller/ViewController.swift]: Without _.main.async_ **configureModel(_)** would execute before network call, causing unwrapping optional value, as a result the app would crash at runtime

Comment (iKisliy) [reply Cinema Journal/Controller/ViewController.swift]: You're not making network call, you preserve array of already objects. If there is segue perfomed — you have an object in array already guaranteed

**Pull request merged**